command/import.go

```diff
@@ -0,0 +1,98 @@
+package command
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	adminserver "github.com/filecoin-project/indexer-reference-provider/server/http"
+	"github.com/urfave/cli/v2"
+)
+
+var ImportCmd = &cli.Command{
+	Name:        "import",
+	Aliases:     []string{"i"},
+	Usage:       "Imp",
+	Subcommands: []*cli.Command{importCarSubCmd},
+}
+
+var (
+	metadata        []byte
+	key             []byte
+	importCarSubCmd = &cli.Command{
+		Name:    "car",
+		Aliases: []string{"c"},
+		Usage:   "Imports CAR from a path",
+		Flags:   importCarFlags,
+		Before:  beforeImportCar,
+		Action:  doImportCar,
+	}
+)
+
+func beforeImportCar(context *cli.Context) error {
+	if metadataFlagValue != "" {
+		decoded, err := base64.StdEncoding.DecodeString(metadataFlagValue)
+		if err != nil {
+			return err
+		}
+		metadata = decoded
+	}
+	if keyFlagValue != "" {
+		decoded, err := base64.StdEncoding.DecodeString(keyFlagValue)
+		if err != nil {
+			return err
+		}
+		key = decoded
+	}
+	return nil
+}
+
+func doImportCar(cctx *cli.Context) error {
+	req := adminserver.ImportCarReq{
+		Path:     carPathFlagValue,
+		Key:      key,
+		Metadata: metadata,
+	}
+
+	reqBody, err := json.Marshal(req)
+	if err != nil {
+		return err
+	}
+	bodyReader := bytes.NewReader(reqBody)
+	httpReq, err := http.NewRequestWithContext(cctx.Context, http.MethodPost, adminAPIFlagValue+"/admin/import/car", bodyReader)
+	if err != nil {
+		return err
+	}
+
+	httpReq.Header.Set("Content-Type", "application/json")
+	cl := &http.Client{}
+	resp, err := cl.Do(httpReq)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to import car, due to failure response from Admin server: %v", http.StatusText(resp.StatusCode))
```

Comment:
will there only ever be an error code and no body of the response with the specific error that you want to relay?

Reply:
On the admin server side there is no error message structure yet. happy to pick that up after this PR is merged to convey more meaningful messages about erroneous cases.